src/components/sbb-journey-summary/sbb-journey-summary.tsx

```diff
@@ -0,0 +1,142 @@
+import { Component, h, JSX, Prop } from '@stencil/core';
+import { InterfaceJourneySummaryAttributes } from './sbb-journey-summary.custom';
+import isTomorrow from 'date-fns/isTomorrow';
+import { isToday, parseISO, format, intervalToDuration, isBefore, isValid } from 'date-fns';
+import {
+  i18nArrival,
+  i18nDeparture,
+  i18nDurationHour,
+  i18nDurationMinute,
+  i18nToday,
+  i18nTomorrow,
+  i18nWalkingDistanceArrival,
+  i18nWalkingDistanceDeparture,
+} from '../../global/i18n';
+import getDocumentLang from '../../global/helpers/get-document-lang';
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-journey-summary.scss',
+  tag: 'sbb-journey-summary',
+})
+export class SbbJourneySummary {
+  @Prop() public summaryConfig!: InterfaceJourneySummaryAttributes['config'];
+
+  private _currentLanguage = getDocumentLang();
+
+  private _formatTime(duration): JSX.Element {
+    return (
+      <span>
+        <time>{duration.hours}</time>
+        <span>{i18nDurationHour.single.short[this._currentLanguage]}</span>
+        <time>{duration.minutes}</time>
+        <span>{i18nDurationMinute.single.short[this._currentLanguage]}</span>
+      </span>
+    );
+  }
+
+  private _renderJourneyStart(departureTime: Date, arrivalTime: Date): JSX.Element {
+    let duration: Duration;
+    if (isBefore(departureTime, arrivalTime))
+      duration = intervalToDuration({ start: departureTime, end: arrivalTime });
+    else {
+      duration =
+        isValid(arrivalTime) && intervalToDuration({ start: arrivalTime, end: departureTime });
+    }
+
+    if (isTomorrow(departureTime) || isToday(departureTime)) {
+      return isTomorrow(departureTime) ? (
+        <span>
+          {i18nTomorrow[this._currentLanguage]}, {this._formatTime(duration)}
+        </span>
+      ) : (
+        <span>
+          {i18nToday[this._currentLanguage]}, {this._formatTime(duration)}
+        </span>
+      );
+    }
+    if (isValid(departureTime))
+      return (
+        <span>
+          <time>{format(departureTime, 'dd.MM, ')}</time>
+          {this._formatTime(duration)}
+        </span>
+      );
+  }
+
+  private _renderJourneyVias(vias: string[]): JSX.Element {
+    const slicedVias = vias.slice(0, 4);
+    return (
+      <div class="journey-summary__via-block">
+        <span class="journey-summary__via-text">Via</span>
+        <ul class="journey-summary__vias">
+          {slicedVias.map((via, index) => (
+            <li class="journey-summary__via">
+              {via}
+              {index !== slicedVias.length - 1 && <span>, </span>}
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+  }
+
+  private _renderWalkTime(before: boolean, duration: number): JSX.Element {
```

Comment:
This would be redundant if we merge the sbb-pearl-chain-time. (But maybe after both tickets are merged)

I think its really better this way